src/main/java/org/elasticsearch/index/query/TermQueryBuilder.java

```diff
@@ -22,178 +22,52 @@
-
-    private String queryName;
-
-    /**
-     * Constructs a new term query.
-     *
-     * @param fieldName  The name of the field
-     * @param value The value of the term
-     */
+public class TermQueryBuilder extends BaseTermQueryBuilder<TermQueryBuilder> {
+    /** @see BaseTermQueryBuilder#BaseTermQueryBuilder(String, String) */
     public TermQueryBuilder(String fieldName, String value) {
-        this(fieldName, (Object) value);
+        super(fieldName, (Object) value);
     }
 
-    /**
-     * Constructs a new term query.
-     *
-     * @param fieldName  The name of the field
-     * @param value The value of the term
-     */
+    /** @see BaseTermQueryBuilder#BaseTermQueryBuilder(String, int) */
     public TermQueryBuilder(String fieldName, int value) {
-        this(fieldName, (Object) value);
+        super(fieldName, (Object) value);
     }
 
-    /**
-     * Constructs a new term query.
-     *
-     * @param fieldName  The name of the field
-     * @param value The value of the term
-     */
+    /** @see BaseTermQueryBuilder#BaseTermQueryBuilder(String, long) */
     public TermQueryBuilder(String fieldName, long value) {
-        this(fieldName, (Object) value);
+        super(fieldName, (Object) value);
     }
 
-    /**
-     * Constructs a new term query.
-     *
-     * @param fieldName  The name of the field
-     * @param value The value of the term
-     */
+    /** @see BaseTermQueryBuilder#BaseTermQueryBuilder(String, float) */
     public TermQueryBuilder(String fieldName, float value) {
-        this(fieldName, (Object) value);
+        super(fieldName, (Object) value);
     }
 
-    /**
-     * Constructs a new term query.
-     *
-     * @param fieldName  The name of the field
-     * @param value The value of the term
-     */
+    /** @see BaseTermQueryBuilder#BaseTermQueryBuilder(String, double) */
     public TermQueryBuilder(String fieldName, double value) {
-        this(fieldName, (Object) value);
+        super(fieldName, (Object) value);
     }
 
-    /**
-     * Constructs a new term query.
-     *
-     * @param fieldName  The name of the field
-     * @param value The value of the term
-     */
+    /** @see BaseTermQueryBuilder#BaseTermQueryBuilder(String, boolean) */
     public TermQueryBuilder(String fieldName, boolean value) {
-        this(fieldName, (Object) value);
+        super(fieldName, (Object) value);
     }
 
-    /**
-     * Constructs a new term query.
-     *
-     * @param fieldName  The name of the field
-     * @param value The value of the term
-     */
+    /** @see BaseTermQueryBuilder#BaseTermQueryBuilder(String, Object) */
     public TermQueryBuilder(String fieldName, Object value) {
-        this.fieldName = fieldName;
-        if (value instanceof String) {
-            this.value = BytesRefs.toBytesRef(value);
-        } else {
-            this.value = value;
-        }
+        super(fieldName, value);
     }
 
     TermQueryBuilder() {
```

Comment:
I remember making this public, I guess you just need to rebase then to get that change ;)